apps/server/lib/lexical/server/code_intelligence/entity.ex

```diff
@@ -0,0 +1,240 @@
+                  {:skip, zipper, {current, found_line?}}
+              end
+
+            _ ->
+              {:cont, zipper, {current, found_line?}}
+          end
+
+        zipper, acc ->
+          {:cont, zipper, acc}
+      end)
+
+    if innermost do
+      {:ok, innermost}
+    else
+      {:error, :not_found}
+    end
+  end
+
+  # Handle aliases explicitly because Sourceror does not expect the atoms
+  # making up the components of the alias to be wrapped in tuples with
+  # additional metadata.
+  defp fetch_range({:__aliases__, _, [_ | _] = args}) do
+    {_, start_meta, _} = List.first(args)
+    {_, end_meta, [atom]} = List.last(args)
+
+    range = %{
+      start: [line: start_meta[:line], column: start_meta[:column]],
+      end: [line: end_meta[:line], column: end_meta[:column] + String.length(to_string(atom))]
+    }
+
+    {:ok, range}
+  end
+
+  # Corrects dot-call issue
+  defp fetch_range({:., _, [left, atom]}) when is_atom(atom) do
+    with {:ok, range} <- fetch_range(left) do
+      # Unhandled edge case: quoted atoms, e.g. Foo."bar-baz"
+      len = atom |> Atom.to_string() |> String.length()
+      {:ok, update_in(range.end[:column], &(&1 + 1 + len))}
+    end
+  end
+
+  # There are certain AST nodes that fail on Sourceror.get_range/1, like
+  # string interpolation segments, for instance. Instead of causing an
+  # error in the server, we just bail out on fetching the range.
+  defp fetch_range(node) do
+    {:ok, Sourceror.get_range(node)}
+  rescue
+    _ ->
+      Logger.warning("Couldn't get range for node: #{inspect(node)}")
+      {:error, :no_range}
+  end
+
+  @doc """
+  Returns the source location of the entity at the given position in the document.
+  """
+  def definition(%Project{} = project, %Document{} = document, %Position{} = position) do
+    maybe_location = RemoteControl.Api.definition(project, document, position)
+    parse_location(maybe_location, document)
+  end
+
+  defp parse_location(%ElixirSense.Location{} = location, document) do
+    %{file: file, line: line, column: column} = location
+    file_path = file || document.path
+    uri = Document.Path.ensure_uri(file_path)
+
+    with {:ok, document} <- Document.Store.open_temporary(uri),
+         {:ok, text} <- Document.fetch_text_at(document, line) do
+      range = to_precise_range(text, line, column)
+
+      {:ok, Location.new(range, document)}
+    else
+      _ ->
+        {:error, "Could not open source file or fetch line text: #{inspect(file_path)}"}
+    end
+  end
+
+  defp parse_location(nil, _) do
+    {:ok, nil}
+  end
+
+  defp to_precise_range(text, line, column) do
+    case Code.Fragment.surround_context(text, {line, column}) do
+      %{begin: start_pos, end: end_pos} ->
+        to_range(start_pos, end_pos)
+
+      _ ->
+        # If the column is 1, but the code doesn't start on the first column, which isn't what we want.
+        # The cursor will be placed to the left of the actual definition.
```

Comment:
does it matter if the column is 1 here? Code that's indented might not start on line 1 either

Reply:
iirc, this comment was copied here from another file during some reorganization. I'll look more carefully at this logic though and make sure that whatever special case is being handled makes sense.